user-interface/src/main/java/life/qbic/datamanager/views/general/CopyToClipBoardComponent.java

```diff
@@ -0,0 +1,145 @@
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.icon.Icon;
+import com.vaadin.flow.component.icon.VaadinIcon;
+import com.vaadin.flow.shared.communication.PushMode;
+import com.vaadin.flow.theme.lumo.LumoUtility.IconSize;
+import java.io.Serial;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Copy to Clipboard Component
+ * <p>
+ * Customizable component based on {@link Span} which enables the user to copy the contained string
+ * into the clipboard. Once the user clicks on the copy icon, the icon is switched for specified
+ * successTimer to an icon indicating the success of the copy process. The switch between the copy
+ * icon and the success icon can be listened to via the {@link SwitchToCopyIconEvent} and
+ * {@link SwitchToSuccessFullCopyIconEvent}
+ */
+@JsModule("./javascript/copytoclipboard.js")
+public class CopyToClipBoardComponent extends Span {
+
+  private static final Logger log = LoggerFactory.getLogger(CopyToClipBoardComponent.class);
+  private final Icon copyIcon;
+  private final Icon copySuccessIcon;
+  private long showSuccessTime = 1000;
+  private String textToBeCopied = "";
+
+  public CopyToClipBoardComponent() {
+    copyIcon = VaadinIcon.COPY_O.create();
+    copyIcon.setSize(IconSize.SMALL);
+    copyIcon.addClassNames("clickable", "copy-icon");
+    copyIcon.setTooltipText("Copy text to clipboard");
+    copySuccessIcon = VaadinIcon.CHECK.create();
+    copySuccessIcon.addClassName("copy-icon-success");
+    add(copyIcon);
+    add(copySuccessIcon);
+    copySuccessIcon.setVisible(false);
+    addClickListener(this::handleCopyClicked);
+  }
+
+  public CopyToClipBoardComponent(String textToBeCopied) {
+    this();
+    setCopyText(textToBeCopied);
+  }
+
+  public void setCopyText(String textToBeCopied) {
+    this.textToBeCopied = textToBeCopied;
+  }
+
+  public void setIconSize(String size) {
+    copyIcon.setSize(size);
+    copySuccessIcon.setSize(size);
+  }
+
+  private void handleCopyClicked(ComponentEvent<Span> componentEvent) {
+    fireEvent(new SwitchToSuccessFullCopyIconEvent(this, componentEvent.isFromClient()));
+    UI.getCurrent().getPage().executeJs("window.copyToClipboard($0)", textToBeCopied);
+    copyIcon.setVisible(false);
+    copySuccessIcon.setVisible(true);
+    // reset copy view after a specific time
+    UI ui = UI.getCurrent();
+    ui.getPushConfiguration().setPushMode(PushMode.MANUAL);
+    new Thread(() -> {
+      try {
+        TimeUnit.MILLISECONDS.sleep(showSuccessTime);
+        fireEvent(new SwitchToCopyIconEvent(this, componentEvent.isFromClient()));
+      } catch (InterruptedException ex) {
+        Thread.currentThread().interrupt();
+        log.debug("CopyToClipboard thread interrupted due to{}", ex.getMessage());
+      }
+      ui.access(() -> {
+        copyIcon.setVisible(true);
+        copySuccessIcon.setVisible(false);
+        ui.push();
+      });
+    }).start();
```

Comment:
The creation of a thread is a very resource expensive task and we should try to avoid it, since we also risk to overload our system with to many open threads. The simplest way is to wrap in a CompletableFuture object, and call its join method. Completable Future use the ForkJoinPool. commonPool(), which we are happy with for the moment.